.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,29 @@
+name: shellcheck
+
+on: [push, pull_request]
+
+jobs:
+  lint_shellscripts:
+    runs-on: [ubuntu-latest]
+    container:
+      image: koalaman/shellcheck-alpine:v0.7.1
+      options: --entrypoint /bin/sh
```

Comment:
Already asked this on the issue #3170 - what's the problem with using the actual GH action?

Reply:
Hey sorry, I missed that comment in the ticket.  I can try to get that GitHub action working if that's a deal-breaker for you all. In previous versions of that marketplace action, I wasn't able to get it to run on another, which is why I chose to use the maintainer of shellcheck's (Koalaman's) container.

In my mind using Koalaman's docker image is a more elegant solution than the [logic outlined](https://github.com/ludeeus/action-shellcheck/blob/master/action.yaml#L59-L66) it the GitHub action (which downloads and installs Koalaman's binaries from GitHub).

As for the security of the image, it is [FROM scratch](https://github.com/koalaman/shellcheck/blob/master/Dockerfile.multi-arch#L22-L26) which would arguable make it much more secure than that Ludeeus' GitHub Action.

At the end of the day, I guess this container preference just arose from building CI steps with containers from running Jenkins on K8s with dynamically rendered pods and GitLab CI.

Reply:
Hi!

I agree that the fact that it is using "FROM SCRATCH" and the Dockerfile is nicely readable makes the Docker Image itself more secure. 
Unfortunately, there is no way of verifying that the Docker Image on DockerHub is actually being build from the Dockerfile provided in the repository. Even if it is possible to verify that **today**, the image might change tomorrow. Or next week. or ...
On DockerHub, you can always delete or overwrite a tag. I'm not saying that Koalaman is going to do that, but it's entirely possible that someone will try to do that (or try to hack an account on DockerHub with that purpose), hence it's better to avoid referring to (non official) images on DockerHub.

FYI, [Ludeeus' shellcheck GitHub Action](https://github.com/ludeeus/action-shellcheck) is a [composite action](https://github.com/ludeeus/action-shellcheck/blob/f0d446b80e91cc0a7e8bec41ada75e2e73fd2218/action.yaml#L40) which does not require a Docker Image to be shipped with it.

In the end, using that action does not make it more secure, but it makes it traceable (plus GH is having an eye on it as it's on marketplace).

Anyway, an alternative way that neither uses the Docker Image on DockerHub nor the GH action is to write the installation/download of the tool in bash, similar to what we did with [pluto and helm](https://github.com/keptn/keptn/blob/master/.github/workflows/deprecated-version-check.yml#L11-L15). 

(Btw, I'm not saying that this is the wrong way of doing it. I used to do a lot of stuff with GitLab CI in the past and we used external Docker Images all the time. Though we had to get permission to use an image, the image was audited and then pulled to our internal registry **once**.)

Reply:
Alright, I just tested that action from the market place and it seems to be pretty interchangeable.